# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1696366047 -7200
# Parent  4168bf9982a664c2ee275a91e0336865c01574c6
Misc: Introduce pref mail.download.open_attachments_readonly.

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -1427,16 +1427,17 @@ pref("betterbird.donation.eoy.url", "htt
 
 // Make this visible in the advanced preferences.
 pref("ui.prefersReducedMotion", 0);
 
 pref("mail.multiselect_message_shows_current", false);
 pref("mail.startupMinimized", false);
 // Avoid .part files on desktop when opening attachments from compose window.
 pref("browser.download.start_downloads_in_tmp_dir", true);
+pref("mail.download.open_attachments_readonly", true);
 
 // IMAP-JS disabled, Bug 1707547.
 pref("mailnews.imap.jsmodule", false);
 
 // Unified toolbar
 
 // 0: icons beside text
 // 1: icons above text
diff --git a/mail/modules/AttachmentInfo.sys.mjs b/mail/modules/AttachmentInfo.sys.mjs
--- a/mail/modules/AttachmentInfo.sys.mjs
+++ b/mail/modules/AttachmentInfo.sys.mjs
@@ -305,17 +305,19 @@ export class AttachmentInfo {
 
         Cc["@mozilla.org/uriloader/external-helper-app-service;1"]
           .getService(Ci.nsPIExternalAppLauncher)
           .deleteTemporaryFileOnExit(tempFile);
 
         await saveToFile(tempFile.path, true);
         // Before opening from the temp dir, make the file read-only so that
         // users don't edit and lose their edits...
-        await IOUtils.setPermissions(tempFile.path, 0o400); // Set read-only
+        if (Services.prefs.getBoolPref("mail.download.open_attachments_readonly", true)) {
+          await IOUtils.setPermissions(tempFile.path, 0o400); // Set read-only
+        }
         this._openFile(mimeInfo, tempFile);
       };
 
       let openLocalFile = mimeInfo => {
         let fileHandler = Services.io
           .getProtocolHandler("file")
           .QueryInterface(Ci.nsIFileProtocolHandler);
 
